app/client/components/homePage/GoogleMap.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react";
+import { Map, Marker, useMap, useMapsLibrary } from "@vis.gl/react-google-maps";
+
+interface GoogleMapProps {
+  address: string;
+}
+
+export const GoogleMap = ({ address }: GoogleMapProps) => {
+  const map = useMap();
+  const placesLib = useMapsLibrary("places");
+
+  const [placesService, setPlacesService] = useState<
+    google.maps.places.PlacesService | undefined
+  >(undefined);
+
+  const [markerPosition, setMarkerPosition] = useState({ lat: 0, lng: 0 });
+  const [isMarkerReady, setIsMarkerReady] = useState(false);
+
+  useEffect(() => {
+    if (!placesLib || !map) return;
+    setPlacesService(new placesLib.PlacesService(map));
+  }, [placesLib, map]);
+
+  useEffect(() => {
```

Comment:
`placesService.findPlaceFromQuery()` sends a network request to Google API, so it has to be inside of a `useEffect()`. 

I've separated `new placesLib.PlacesService(map)` to a separate `useEffect()`, because that's how they do it in the [docs](https://visgl.github.io/react-google-maps/docs/guides/interacting-with-google-maps-api#other-maps-javascript-api-libraries-and-services). I guess that the idea is that you should import the `places` library only once and then use it many times in another `useEffect()`. But the address we search for will never change on the frontend side, so I moved it now to the same `useEffect()` in order to avoid additional rerender, just like you mentioned.